index.js

```diff
@@ -85,263 +86,251 @@ class LatticeKeyring extends EventEmitter {
+            txData.accessList = tx.accessList || [];
+            txData.type = 2;
+            break;
+          case 1: // eip2930
+            txData.accessList = tx.accessList || [];
+            txData.gasPrice = `0x${tx.gasPrice.toString('hex')}`;
+            txData.type = 1;
+            break;
+          default: // legacy
+            txData.gasPrice = `0x${tx.gasPrice.toString('hex')}`;
+            txData.type = null;
+            break;
+        }
+        // Lattice firmware v0.11.0 implemented EIP1559 and EIP2930
+        // We should throw an error if we cannot support this.
+        if (fwVersion.major === 0 && fwVersion.minor <= 11 && txData.type) {
+          throw new Error('Please update Lattice firmware.');
         }
+      } catch (err) {
+        throw new Error(`Failed to build transaction.`)
+      }
+      // Get the signature
+      signedTx = await this.sdkSession.sign({ currency: 'ETH', data: txData });
 
-        // Build the tx for export
-        let validatingTx;
-        const _chainId = `0x${this._getEthereumJsChainId(tx).toString('hex')}`;
-        const chainId = new BN(_chainId).toNumber();
-        const customNetwork = Common.forCustomChain('mainnet', {
-          name: 'notMainnet',
-          networkId: chainId,
-          chainId: chainId,
-        }, 'london')
-
-        validatingTx = EthTx.TransactionFactory.fromTxData(txToReturn, {
-          common: customNetwork, freeze: Object.isFrozen(tx)
-        })
-        return resolve(validatingTx);
-      })
-      .catch((err) => {
-        return reject(new Error(err));
-      })
+      // Add the sig params. `signedTx = { sig: { v, r, s }, tx, txHash}`
+      if (!signedTx.sig || !signedTx.sig.r || !signedTx.sig.s) {
+        throw new Error('No signature returned.');
+      }
+      if (signedTx.sig.v === undefined) {
+        // V2 signature needs `v` calculated
+        v = getV(tx, signedTx);
+      } else {
+        // Legacy signatures have `v` in the response
+        v = signedTx.sig.v.length === 0 ? '0' : signedTx.sig.v.toString('hex')
+      }
+    }
+
+    // Pack the signature into the return object
+    txToReturn.r = Util.addHexPrefix(signedTx.sig.r.toString('hex'));
+    txToReturn.s = Util.addHexPrefix(signedTx.sig.s.toString('hex'));
+    txToReturn.v = Util.addHexPrefix(v);
+
+    // Make sure the active wallet is correct to avoid returning
+    // a signature from an unexpected signer.
+    const foundIdx = await this._accountIdxInCurrentWallet(address);
+    if (foundIdx === null) {
+      throw new Error(
+        'Wrong account. Please change your Lattice wallet or ' +
+        'switch to an account on your current active wallet.'
+      );
+    }
+    return EthTx.TransactionFactory.fromTxData(txToReturn, {
+      common: tx.common, freeze: Object.isFrozen(tx)
     })
+    return tx;
   }
 
-  signPersonalMessage(address, msg) {
+  async signPersonalMessage(address, msg) {
```

Comment:
This doesn't need to be marked as async, since it's returning a promise anyway.

Reply:
Need to stick with keyring API